lib/utils.js

```diff
@@ -77,16 +77,34 @@ function waitForAvailable(host, port, options, callback) {
       count--;
       if (available) return callback();
       if (count === 0) return callback(new Error('Server is unavailable'));
-      setTimeout(() => run(), options.retryMs);
+      setTimeout(() => run(), options.retryMS);
     });
   }
 
   run();
 }
 
+function createCommandOptions(manager, localOptions) {
+  const options = clone(manager.clientOptions);
```

Comment:
can we use `Object.assign` instead of `clone`? We should aim to completely remove `clone` in the future.

Reply:
`Object.assign` didn't exist before ES6, so people used to implement something like `clone` in every project. The implementation of `clone` here is like a one dimensional implementation of `Object.assign`, where it doesn't support multiple applied objects.